### ISSUE TITLE
TrustSitelists as a list

### ISSUE THREAD

**Issue by vlimant** [dmwm/WMCore#9059]
what are the technical implications of allowing the TrustSitelists to be a list of sites of sites.

TrustSitelists = False means enforce data location for job site white list.

TrustSitelists = True and TrustSitelists = [] are equivalent and discard totally the data location for job site white list, using only SiteWhitelist.

TrustSitelists = ["T1_IT_CNAF"] means for things located at storage link to CNAF, use SiteWhitelist instead of enforcing data location, for things located elsewhere, enforce data location.

Would nicely solve https://github.com/CMSCompOps/WmAgentScripts/issues/380, and can be used otherwise (remember #5864 ?)

**Comment by amaltaro** [comment]
Writing it in different words, just to make sure I understand what you're asking for. The desired behaviour would be as:
* `TrustSitelists = []`: do not trust anything and run the usual data location matching with the SiteWhitelist.
* `TrustSitelists = ["T1_IT_CNAF"]`: run the data location mapping for the whole input data, then
  * if the data is placed at CNAF (sites present in this TrustSitelists), force the SiteWhitelist as their data location;
  * else (data located somewhere else), use the usual data location matching with the SiteWhitelist

Is that correct? If so, what's expected for the following two scenarios:
1) I want to force my SiteWhitelist as location no matter where the input data is. Ideally we could set it to `True`, but then we make a mess of this parameter where it can be different data types (boolean and list). Thus, you'd have to set this variable to ALL sites existent in the production environment...
2) what if an input block is placed at multiple locations, and one of those is the one you defined in the `TrustSitelists`? Should we force the SiteWhitelist to that block/WQE or should we run the usual data location + SiteWhitelist matching?

**Comment by vlimant** [comment]
> TrustSitelists = True and TrustSitelists = [] are equivalent and discard totally the data location for job site white list, using only SiteWhitelist.

yes, having TrustSitelists with possible multiple type is not pleasant ; you could include a SiteForTrustSitelists of list type be use when TrustSitelists = True.

**Comment by vlimant** [comment]
> TrustSitelists = ["T1_IT_CNAF"] means for things located at storage link to CNAF, use SiteWhitelist instead of enforcing data location, for things located elsewhere, enforce data location.

meaning that for anything located in a site within TrustSitelists (or SiteForTrustSitelists), use SiteWhitelist instead of enforcing data location, for things located elsewhere, enforce data location
